model/common/src/icon4py/model/common/test_utils/serialbox_utils.py

```diff
@@ -1688,14 +1782,15 @@ def vz0r(self):
 
 
 class IconSerialDataProvider:
-    def __init__(self, fname_prefix, path=".", do_print=False, mpi_rank=0):
+    def __init__(self, fname_prefix, path=".", do_print=False, mpi_rank=0, advection=False):
         self.rank = mpi_rank
         self.serializer: serialbox.Serializer = None
         self.file_path: str = path
         self.fname = f"{fname_prefix}_rank{self.rank!s}"
         self.log = logging.getLogger(__name__)
         self._init_serializer(do_print)
-        self.grid_size = self._grid_size()
+        if not advection:  # TODO (dastrm): somebody should make this class load only what it needs
+            self.grid_size = self._grid_size()
```

Comment:
How do cap the size of the fields then? This is used when reading in fields  to reduce the field size in the savepoints to the actual sizes of each dimension. The current implementation assumes that there is a grid_savepoint but the `_get_field` in the `IconSavepoint` uses this size.